### PR TITLE
Add a sniff to detect the usage of the backtick operator.

### PR DESCRIPTION
**Pull request by jrfnl**
Working on a project now which wants to add a sniff like this as part of their security-sniffs and figured it would be re-usable for others as well.

### CODE REVIEW

File: CodeSniffer/Standards/Generic/Tests/PHP/BacktickOperatorUnitTest.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+     * The key of the array should represent the line number and the value
+     * should represent the number of errors that should occur on that line.
+     *
+     * @return array<int, int>
+     */
+    public function getErrorList()
+    {
+        $errors = array();
+
+        if ($this->isSafeModeEnabled() === false) {
```
Comment (aik099) [inline CodeSniffer/Standards/Generic/Tests/PHP/BacktickOperatorUnitTest.php]: Instead you can:

1. also expect errors
2. add `shouldSkipTest` function that would call `$this->isSafeModeEnabled() === false`
3. in fact you can inline the `isSafeModeEnabled` function into `shouldSkipTest`

Comment (jrfnl) [inline CodeSniffer/Standards/Generic/Tests/PHP/BacktickOperatorUnitTest.php]: Good point - adjusting now.

Comment (jrfnl) [inline CodeSniffer/Standards/Generic/Tests/PHP/BacktickOperatorUnitTest.php]: Originally I intended to check for the backtick character in other tokens when safe mode was enabled and throw a warning, but decided against it. This code was left behind from that.

**Comment by gsherwood** [comment]
I ended up making the error message say "forbidden" instead of "strongly discouraged" because it was an error instead of a warning, and it also matches other error messages. But the message and type (error/warning) can be overridden in a ruleset.xml file, so I don't think that should be a problem.

Thanks for the new sniff.

**Comment by jrfnl** [comment]
@gsherwood That's absolutely fine by me and shouldn't be an issue for the depending projects (in my case).

**Pull request merged**